lib/pdk/generators/module.rb

```diff
@@ -77,63 +78,98 @@ def self.prepare_module_directory(target_dir)
       end
 
       def self.module_interview(metadata, opts = {})
-        puts _(
-          'We need to create a metadata.json file for this module. Please answer the ' \
-          'following questions; if the question is not applicable to this module, feel free ' \
-          'to leave it blank.',
-        )
-
-        begin
-          puts ''
-          forge_user = PDK::CLI::Input.get(_('What is your Puppet Forge username?'), metadata.data['author'])
-          metadata.update!('name' => "#{forge_user}-#{opts[:name]}")
-        rescue StandardError => e
-          PDK.logger.error(_("We're sorry, we could not parse your module name: %{message}") % { message: e.message })
-          retry
-        end
+        questions = [
+          {
+            name:             'name',
+            question:         _('What is your Puppet Forge username?'),
+            help:             _('This will be used when uploading your module to the Forge. You can opt out of this at any time.'),
+            required:         true,
+            validate_pattern: %r{\A[a-z0-9]+\Z},
+            validate_message: _('Forge usernames can only contain letters and numbers'),
```

Comment:
Yeah, I just stole that pattern from `PDK::Module::Metadata`'s name validation :)